server/datastore/mysql/software.go

```diff
@@ -659,56 +659,45 @@ func (si *softwareIterator) Next() bool {
 	return si.rows.Next()
 }
 
-func (ds *Datastore) ListSoftwareBySourceIter(
+// AllSoftwareIterator Returns an iterator for the 'software' table, filtering out
+// software entries based on the 'query' param.
+func (ds *Datastore) AllSoftwareIterator(
 	ctx context.Context,
-	sources []string,
+	query fleet.SoftwareIterQueryOptions,
 ) (fleet.SoftwareIterator, error) {
-	if len(sources) == 0 {
-		return nil, errors.New("please provide at least one source")
-	}
-
 	var err error
 	var args []interface{}
 
 	stmt := `SELECT 
-		s.id, 
-		s.name,
-		s.version,
-		s.bundle_identifier,
-		s.release,
-		s.vendor,
-		s.arch,
-		s.source 
-	FROM software s
-	WHERE source IN (?)`
-
-	stmt, args, err = sqlx.In(stmt, sources)
+		s.* ,
+		COALESCE(sc.cpe, '') AS generated_cpe
+	FROM software s 
+	LEFT JOIN software_cpe sc ON (s.id=sc.software_id)`
+	// The rows.Close call is done by the caller once iteration using the
+	// returned fleet.SoftwareIterator is done.
```

Comment:
Let's add this to the method documentation.

Reply:
Done in https://github.com/fleetdm/fleet/pull/10514/commits/8c2e90168b77aaf402e96e4b18c5e5bc2ce018c8